sdk/dotnet/Pulumi/Resources/DependencyResource.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// <see cref="DependencyResource"/> is a <see cref="Resource"/> that is used to indicate that an
+    /// <see cref="Output"/> has a dependency on a particular resource. These resources are only created when dealing
+    /// with remote component resources.
+    /// </summary>
+    public sealed class DependencyResource : CustomResource
```

Comment:
This is an excellent question. It looks like this could be internal and not exposed to users, at least for now.

@pgavlin, can you think of any reason this needs to be public right now? If not, I'll plan to make it (and `DependencyProviderResource`) internal for the time being. And will look into doing the same for TypeScript and Python.

Reply:
Updated the PR to not expose this publicly.